src/parser.rs

```diff
@@ -125,6 +141,32 @@ pub fn inline<T>(result: Result<T, ParserError>) -> Result<T, (ParserError, Toke
 	result.map_err(|error| (error, Token::Terminator))
 }
 
+pub fn animation(lexer: &mut Lexer, check_with: bool) -> Result<Option<AnimationDeclaration>, (ParserError, Token)> {
+	if check_with {
+		match inline(lexer.token())? {
+			None | Some(Token::Terminator) => return Ok(None),
+			Some(Token::Identifier(string)) if string == "with" => {}
```

Comment:
Use `()` instead of `{}` for empty statements. Be consistent in your variable names too. As it's an identifier you should use `identifier` instead of `string`.

Reply:
Done. The reason it was `string` was because I initially thought Identifier and String were the same thing token wise and only changed it at the end.